### PR TITLE
[New arch] Sync folder recursively

### PR DESCRIPTION
**Pull request by abelgardep**
## Related Issues
Implements #3707 
_____

### CODE REVIEW

**Comment by abelgardep** [comment]
### Current state
Syncs recursively all content when clicking on the circle arrow over a folder.
Refresh folder syncs only already downloaded files within this folder, not recursively

### Known issues
1. Folder synchronization can't be cancelled at the moment. Uploads and Downloads are enqueued but can't be cancelled all at the same time
2. Refresh folder when picking a folder (move/copy/camera uploads path) triggers syncs for already downloaded files in the folder (and probably for av offline files in the future). Not sure if this is an issue, but it may takes longer than expected. Just a refresh could do the work when picking a folder

File: owncloudApp/build.gradle [hunk 1]
```diff
@@ -93,7 +93,6 @@ dependencies {
 
     debugImplementation "androidx.fragment:fragment-testing:1.4.1"
     debugImplementation 'com.facebook.stetho:stetho:1.6.0'
-    debugImplementation 'com.squareup.leakcanary:leakcanary-android:2.9.1'
```
Comment (JuancaG05) [inline owncloudApp/build.gradle:96]: Was this intentional? 🤨 It also bothers me sometimes, but I don't know if you wanted to remove it definitely 😀

**Review by JuancaG05 — CHANGES_REQUESTED** [review]
Nice improvement @abelgardep! 💯 Much better to send a String (`accountName`) than the whole account when it's not necessary 😀
Just a tiny comment about something you removed, well done!

**Review by JuancaG05 — APPROVED** [review]
LGTM 👍

**Pull request merged**